Updates to slack addon (#218)
ynput/ayon-documentation: website/docs/addon_slack.md

import TabItem from '@theme/TabItem';

This addon enables you to set up profiles that specify when to trigger notifications, and for which combination of task, host, and product types.
It also supports customizable message templates that use placeholders (denoted by curly brackets `{}`) that are automatically populated at the time of publishing.

## App installation

> Please ensure that the Slack application is installed on the company's Slack before proceeding.

Reviewer: I got quite confused reading over this page because it leaves a bit in the middle which section applies to Slack and which applies to AYON.

I wonder if phrasing it like this helps clarify what part of the installation is "slack" related and when you're moving to AYON specifics.
```suggestion
Please ensure that the AYON Slack application is installed on the company's Slack using the `manifest.yml` before proceeding to AYON settings.
```

2. To create your app, please follow the steps outlined in the "Creating apps with manifests" section at [Slack API Documentation](https://api.slack.com/reference/manifests#using).

:::tip
You can quickly navigate to the local app data using the environment variable shortcut: `%LOCALAPPDATA%`. For example, using `%LOCALAPPDATA%\Ynput\AYON\addons` as path to see your installed addons.
:::

</TabItem>

<TabItem value="linux">

1. Locate the `manifest.yml` file. You can find it in the user data folder, typically at `~/.local/share/Ynput/AYON/addons/slack_X.X.X/ayon_slack/`.
2. To create your app, please follow the steps outlined in the "Creating apps with manifests" section at [Slack API Documentation](https://api.slack.com/reference/manifests#using).

</TabItem>

<TabItem value="mac">

1. Locate the `manifest.yml` file. You can find it in the `Application Support` folder, typically at `~/Library/Application Support/Ynput/AYON/addons/slack_X.X.X/ayon_slack/`.
2. To create your app, please follow the steps outlined in the "Creating apps with manifests" section at [Slack API Documentation](https://api.slack.com/reference/manifests#using).

</TabItem>

</Tabs>

### App icon

> To give the bot a custom icon, the Slack admin needs to upload it following the app installation.

Reviewer: Here I kind of have the same issue - "following the app installation" may be a bit confusing:

- Is that the Slack app installation using the manifest? I believe it's this.
- Is that the `ayon-slack` addon installation to AYON?

Would be great if we can find some wording that may bring less ambiguity.

Reviewer 2: IMO, I think it's a matter of context. 
For instance, I assumed that the person who is reading this page and want to configure his/her/whatsoever slack settings is already aware of the app installation is related to a slack app.  Also, the instructions are referring to their slack website/instance.

Reviewer 2: Let me know if this helps. Feel free to unresolve.
https://github.com/ynput/ayon-documentation/pull/218/commits/318a33c5df7ffdbad3b85e03b8ac5f4537262355

Here's how to do it:

1. Navigate to your Slack app settings at a URL similar to `https://api.slack.com/apps/XXXXXXXX/general`.
2. Click on 'Basic Information' and then find 'Display Information'.

> 3. Here, you can upload any image of your choice. For ease, you can use the pre-made AYON icon available at [this link](https://github.com/ynput/ayon-launcher/blob/develop/common/ayon_common/resources/AYON.png).

Reviewer: ```suggestion
3. Here, you can upload any image of your choice. For ease, you can use the [pre-made AYON icon](https://github.com/ynput/ayon-launcher/blob/develop/common/ayon_common/resources/AYON.png).
```

  - `ayon+settings://slack/publish/CollectSlackFamilies/enabled?project={PROJECT_NAME}`

### Token

![](assets/slack/auth_key.png)

For the Slack addon to function properly, it's essential to enter the authentication token:

- For all projects:
  - `ayon+settings://slack/token`
- For a specific project, replace {PROJECT_NAME} with your project's name:
  - `ayon+settings://slack/token?project={PROJECT_NAME}`

You'll find this token in the Slack dashboard once the app is installed. Remember, you can create separate tokens for use with different projects.

![Get token](assets/slack/slack_token.png)

### Profiles

![](assets/slack/profiles.png)

Profiles determine when notifications are triggered. You can configure one or several profiles based on `Product types`, `Hosts`, `Task types`, and `Task names` or `Product names` (with regex support).

For example, to receive notifications when a render is published from Maya, the settings would be:

- Product type: 'render'
- Host: 'Maya'

### Messages to channels

![Configure module](assets/slack/messages_to_channels.png)

- **Channels:**
Specify one or more channels where you'd like the plugin to send messages.

> :::info
> Please note that the Slack bot can access public channels by default, so there's no need for it to join them first.
> :::
> - **Upload thumbnail** & **Upload review:**
> The plugin is capable of uploading a 'thumbnail' or 'review' from an instance (if they are present in an instance). However, your Slack bot must be manually invited to each target channel by a Slack admin.
> :::tip
> To add your bot to a target channel, simply mention it in the message field with the command: `/invite @ayonNotifier`.
> :::

Reviewer: The info message here and the tip here seem to be very related to the same thing.

I wonder if it makes more sense to combine them, by writing e.g.

```suggestion

- **Upload thumbnail** & **Upload review:**
The plugin is capable of uploading a 'thumbnail' or 'review' from an instance (if they are present in an instance). However, your Slack bot must be manually invited to each target channel by a Slack admin if they are private channels.
:::note
Please note that the Slack bot can access public channels by default, so there's no need for it to join them first.
To add your bot to a private target channel, simply mention it in the message field with the command: `/invite @ayonNotifier` inside that channel.
:::
```

Reviewer 2: Let me know if this made it better.
https://github.com/ynput/ayon-documentation/pull/218/commits/b91c4749cfb3e59fddc3a7a933310f161ebfda47

- **Message:**
Message content can use template keys (see [Available template keys](admin_settings_project_anatomy#available-template-keys)).
Few keys also have Capitalized and UPPERCASE format. Values will be modified accordingly. e.g. `{folder[name]}` ➜ "Gun", `{product}` ➜ "RENDER".

> :::tip

Reviewer: This is lacking examples now of "Capitalized" and "UPPERCASE" format?

> Specify one or more channels where you'd like the plugin to send messages.
> :::info
> Please note that the Slack bot can access public channels by default, so there's no need for it to join them first.
> :::
> - **Upload thumbnail** & **Upload review:**
> The plugin is capable of uploading a 'thumbnail' or 'review' from an instance (if they are present in an instance). However, your Slack bot must be manually invited to each target channel by a Slack admin.
> :::tip
> To add your bot to a target channel, simply mention it in the message field with the command: `/invite @ayonNotifier`.
> :::
> - **Message:**
> Message content can use template keys (see [Available template keys](admin_settings_project_anatomy#available-template-keys)).
> Few keys also have Capitalized and UPPERCASE format. Values will be modified accordingly. e.g. `{folder[name]}` ➜ "Gun", `{product}` ➜ "RENDER".
> :::tip
> We have one additional implemented key `{review_filepath}`
> :::

Reviewer: ```suggestion
Specify one or more channels where you'd like the plugin to send messages.

:::info
Please note that the Slack bot can access public channels by default, so there's no need for it to join them first.
:::

- **Upload thumbnail** & **Upload review:**
The plugin is capable of uploading a 'thumbnail' or 'review' from an instance (if they are present in an instance). However, your Slack bot must be manually invited to each target channel by a Slack admin.

:::tip
To add your bot to a target channel, simply mention it in the message field with the command: `/invite @ayonNotifier`.
:::

- **Message:**
Message content can use template keys (see [Available template keys](admin_settings_project_anatomy#available-template-keys)).
Few keys also have Capitalized and UPPERCASE format. Values will be modified accordingly. e.g. `{folder[name]}` ➜ "Gun", `{product}` ➜ "RENDER".

:::tip
We have one additional implemented key `{review_filepath}`
:::
```

Cosmetics...

It would be nice to maybe provide an example as to what `{review_filepath}` resolves to? Is it the published path from the machine emitting the signal? (this may be nice info for e.g. studios using site-sync setups) or is it a rootless path, etc?

Reviewer 2: I've added an example here https://github.com/ynput/ayon-documentation/pull/218/commits/b91c4749cfb3e59fddc3a7a933310f161ebfda47 (reused an existing example),
I'm not sure how it works with sitesync or rootless paths. maybe @kalisp can help providing this example/info.

Reviewer: Feel free to resolve if you feel this has been fixed. 👍
